text/0006-dataset_naming.md

```diff
@@ -0,0 +1,97 @@
+- Feature Name: dataset_naming
+- Start Date: 
+- RFC PR: 
+- Issue: NA
+
+# Summary
+[summary]: #summary
+
+Define the qri naming system and conventions for name resolution.
+
+# Motivation
+[motivation]: #motivation
+
+Why are we doing this? What use cases does it support? What is the expected outcome?
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+It’s possible to refer to a dataset in a number of ways. It’s easiest to look 
+at the full definition of a dataset reference, and then show what the “defaults” 
+are to make sense of things. The full definition of a dataset reference is 
+as follows:
+
+    dataset_reference = peer_name/dataset_name@/network/hash
+
+an example of that looks like this:
+
+    b5/comics@QmejvEPop4D7YUadeGqYWmZxHhLc4JBUCzJJHWMzdcMe2y/ipfs/QmejvEPop4D7YUadeGqYWmZxHhLc4JBUCzJJHWMzdcMe2y
+
+In a sentence: b5 is the name of a peer, who has a dataset named comics, and 
```

Comment:
Perhaps this belongs in a different RFC, but I would expect / appreciate having definitions here giving exact terms for the sections of this dataset_reference. I say perhaps because I'm not sure we've landed on the exact terms to use everywhere.

For example:
peer_name = b5, the name of the peer that created this dataset
...

Also, the hash Qme...Me2y appears twice with no mention of why.

Reply:
I propose:

peername/dataset_name@peer_id/network/dataset_id

As names.

Could also go with peer_hash and dataset_hash, or id and hash, respectively.

The hash Qme...Me2y appearing twice is a mistake, the peer ID and dataset ID should be different.

Reply:
I'm a fan of peer_id and dataset_id. This layer should theoretically be content-addressing agnostic. dataset naming should rely on unique IDs, which is one of a few properties of content-addressing  / identify-by-hash systems.

Reply:
Any feelings about referring to the final component as "version_id", since it's specifying the exact version of the dataset? All the rest of the names are good to me, I just think it would be too confusing to have both "dataset_name" and "dataset_id".

Right now, the code in repo/ref is using ProfileID to refer to the component after the '@' sign but before the network protocol. I'm fine with changing this to PeerID instead. With peers, there's a direct one-to-one correlation between peername and peerID, whereas datasets using a one-many relationship between dataset_name and version_ids.